src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -402,11 +402,18 @@ void MergeLibraryManifest (string mergedManifest)
 			}
 		}
 
+		public IEnumerable<XElement> ResolveDuplicates (IEnumerable<XElement> elements)
+		{
+			foreach (var e in elements)
+				foreach (var d in ResolveDuplicates (e.Elements ()))
+					yield return d;
+			foreach (var d in elements.GroupBy (x => x.ToFullString ()).SelectMany (x => x.Skip (1)))
```

Comment:
I don't understand how this method works. :-(

*This* part appears to be identical to what's being removed, i.e. this fragment is "buggy" (hence it needing to be fixed in the first place). Presumably then, the *above* `foreach (var e in elements)` loop *fixes* the bug, but I don't understand how these separate loops interact.

Assuming that the `foreach (var e in elements)` ultimately yield 0 elements, we'd still have *this* loop, which is known to be buggy. If the `foreach (var e in elements)` *does* yield elements, that's just *more* elements to remove!

How's this work? It's blowing my mind. :-(

Reply:
The previous method used `Descendants()` which gets every single child element not just those that are immediate children. So given
```xml
<foo>
   <bar>
     <bug/>
   </bar>
   <dar>
     <bug/>
   </dar>
   <too/>
   <too/>
</foo>
```
In the previous method both `bug` elements would be removed because Descendants() gets every element including all children. `Elements()` gets the elements that are children for this exact parent. So the new system will remove duplicates which exist at the same level and ONLY those that exist at the same level. So in the sample above the `bug` will be left as they are as they have different parent elements. But the `too` items will be removed as they have the same parent.